### PR TITLE
[CELEBORN-1490][CIP-6] Support process large buffer in flink hybrid shuffle

### PR DESCRIPTION
**Pull request by reswqa**
### What changes were proposed in this pull request?

This is the last PR in the CIP-6 series.

Fix the bug when hybrid shuffle face the buffer which large then 32K.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java [hunk 1]
```diff
@@ -157,6 +158,27 @@ public void close() {
...
     try {
+      if (byteBuf instanceof CompositeByteBuf) {
+        // If the received byteBuf is a CompositeByteBuf, it indicates that the byteBuf originates
+        // from the Flink hybrid shuffle integration strategy. This byteBuf consists of two parts: a
+        // celeborn header and a data buffer.
+        CompositeByteBuf compositeByteBuf = (CompositeByteBuf) byteBuf;
+        ByteBuf headerBuffer = compositeByteBuf.component(0).unwrap();
+        ByteBuf dataBuffer = compositeByteBuf.component(1).unwrap();
+        dataBuffer.retain();
+        Utils.checkState(dataBuffer instanceof Buffer, "Illegal buffer type.");
```
Comment (SteNicholas) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java:169]:
```suggestion
        Utils.checkState(dataBuffer instanceof Buffer, "Illegal data buffer type for CompositeByteBuf.");
```

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/network/TransportFrameDecoderWithBufferSupplier.java [hunk 2]
```diff
@@ -44,17 +46,37 @@ public class TransportFrameDecoderWithBufferSupplier extends ChannelInboundHandl
   private final ByteBuf msgBuf = Unpooled.buffer(8);
   private Message curMsg = null;
   private int remainingSize = -1;
+  private int totalReadBytes = 0;
+  private int largeBufferHeaderRemainingBytes = -1;
+  private boolean isReadingLargeBuffer = false;
+  private ByteBuf largeBufferHeaderBuffer;
+  public static int DISABLE_LARGE_BUFFER_SPLIT_SIZE = -1;
```
Comment (SteNicholas) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/network/TransportFrameDecoderWithBufferSupplier.java:53]:
```suggestion
  public static final int DISABLE_LARGE_BUFFER_SPLIT_SIZE = -1;
```

File: client-flink/flink-1.20/src/main/java/org/apache/celeborn/plugin/flink/tiered/CelebornTierProducerAgent.java [hunk 3]
```diff
@@ -325,9 +326,18 @@ private void handshake() throws IOException {
     try {
       int remainingReviveTimes = maxReviveTimes;
       while (remainingReviveTimes-- > 0 && !hasSentHandshake) {
+        // In the Flink hybrid shuffle integration strategy, the data buffer sent to the Celeborn
+        // workers consists of two components: the Celeborn header and the data buffers.
+        // In this scenario, the maximum byte size of the buffer received by the Celeborn worker is
+        // equal to the Flink buffer size plus the size of the Celeborn header.
```
Comment (SteNicholas) [inline client-flink/flink-1.20/src/main/java/org/apache/celeborn/plugin/flink/tiered/CelebornTierProducerAgent.java:332]:
```suggestion
        // equal to the sum of the Flink buffer size and the Celeborn header size.
```

**Review by SteNicholas — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Thanks. merge to main(v0.6.0)

**Pull request closed without merging**